sphinx/config.py

```diff
@@ -222,7 +222,7 @@ def convert_overrides(self, name, value):
                                  (name, name + '.key=value'))
             elif isinstance(defvalue, list):
                 return value.split(',')
-            elif isinstance(defvalue, int):
+            elif isinstance(defvalue, int) or name == 'latex_use_xindy':
```

Comment:
This solves the problem but only for this special case.

Reply:
-1: hard coding is not extensible. I suppose there are more boolean settings in another extension. How about the way like #6908 instead?